pkg/appgw/requestroutingrules_test.go

```diff
@@ -311,6 +311,42 @@ var _ = Describe("Test routing rules generations", func() {
 		})
 	})
 
+	Context("test waf policy is not configured in rule path when ingress defined without rules", func() {
+		configBuilder := newConfigBuilderFixture(nil)
+		secret := tests.NewSecretTestFixture()
+		endpoint := tests.NewEndpointsFixture()
+		service := tests.NewServiceFixture(*tests.NewServicePortsFixture()...)
+		ingress := tests.NewSingleServiceIngressFixture()
+		ingress.Annotations[annotations.FirewallPolicy] = "/sub/waf"
+
+		_ = configBuilder.k8sContext.Caches.Secret.Add(secret)
+		_ = configBuilder.k8sContext.Caches.Endpoints.Add(endpoint)
+		_ = configBuilder.k8sContext.Caches.Service.Add(service)
+		_ = configBuilder.k8sContext.Caches.Ingress.Add(ingress)
+
+		cbCtx := &ConfigBuilderContext{
+			IngressList:           []*v1beta1.Ingress{ingress},
+			ServiceList:           []*v1.Service{service},
+			DefaultAddressPoolID:  to.StringPtr("xx"),
+			DefaultHTTPSettingsID: to.StringPtr("yy"),
+		}
+
+		_ = configBuilder.BackendHTTPSettingsCollection(cbCtx)
+		_ = configBuilder.BackendAddressPools(cbCtx)
+		_ = configBuilder.Listeners(cbCtx)
+
+		// !! Action !! -- will mutate pathMap struct
+		pathMap := configBuilder.getPathMaps(cbCtx)
+
+		listenerID := generateListenerID(ingress, nil, n.HTTP, nil, false)
+		It("has no waf policy in pathRule", func() {
+			prs := pathMap[listenerID].ApplicationGatewayURLPathMapPropertiesFormat
+			for _, r := range *prs.PathRules {
```

Comment:
1) Can you also check that the urlPathMap has the correct value for backend pool and backend http setting ?
2) Can you also add another test for where ingress has both rules and backend ?

Reply:
> Can you also check that the urlPathMap has the correct value for backend pool and backend http setting ?
> Can you also add another test for where ingress has both rules and backend ?
Functional test might be a better place to verify if the appgw config generated as expected